src/script/signingprovider.cpp

```diff
@@ -106,28 +111,33 @@ bool FillableSigningProvider::GetPubKey(const CKeyID &address, CPubKey &vchPubKe
         return false;
     }
     auto pubkey = key.GetPubKey();
-    CKey ethKey;
-    const auto ethID = pubkey.GetEthID();
-    // Special case. Bech32 request from a converted Eth address.
-    if (!pubkey.IsCompressed() && ethID != address && GetEthKey(ethID, ethKey)) {
+    if (!pubkey.IsCompressed() && address.type == KeyAddressType::COMPRESSED) {
         pubkey.Compress();
-        vchPubKeyOut = pubkey;
-    } else {
-        vchPubKeyOut = pubkey;
+    } else if (pubkey.IsCompressed() && address.type == KeyAddressType::UNCOMPRESSED) {
+        pubkey.Decompress();
     }
+    vchPubKeyOut = pubkey;
     return true;
 }
 
-bool FillableSigningProvider::AddKeyPubKey(const CKey& key, const CPubKey &pubkey, const CPubKey &compressedPubKey)
+bool FillableSigningProvider::AddKeyPubKey(const CKey& key, const CPubKey &pubkey)
 {
     LOCK(cs_KeyStore);
+
     mapKeys[pubkey.GetID()] = key;
-    if (compressedPubKey.IsValid()) {
-        mapKeys[pubkey.GetEthID()] = key;
-        mapEthKeys[pubkey.GetEthID()] = key;
-        mapKeys[compressedPubKey.GetID()] = key;
+
+    auto pubkeyCopy = pubkey;
```

Comment:
Any specific reason to not just unconditionally store all 3 for easier maintainability?

Reply:
We do store all three. Check the preceding line...

`mapKeys[pubkey.GetID()] = key;`

Reply:
We then store whatever two was not already added below.